sound/soc/sof/intel/hda.c

```diff
@@ -288,8 +296,12 @@ bool hda_common_check_sdw_irq(struct snd_sof_dev *sdev)
 
 static bool hda_dsp_check_sdw_irq(struct snd_sof_dev *sdev)
 {
+	u32 interface_mask = hda_get_interface_mask(sdev);
 	const struct sof_intel_dsp_desc *chip;
 
+	if (!(interface_mask & BIT(SOF_DAI_INTEL_ALH)))
```

Comment:
these change stand out from the rest of the patch. Can it be separated out?

Reply:
they can, but for existing platforms this boils down to if (sdw->dsplessmode)

Reply:
can you elaborate this a bit? It isnt lear to me how this boils down to the same check

Reply:
see https://github.com/thesofproject/linux/blob/9b5c7765a81bbd778e2c7b7a32dbe681c00b51d8/sound/soc/sof/intel/hda.c#L77

The interface mask depends on the value of sdev->dspless

Reply:
@plbossart then does it makes more sense to not enable SDW interrupts, check for SDW interrupt etc if the interface mask is not set? That would be a lot more intuitive than this

Reply:
I think it's more confusing if we add all kinds of tests in common code. It's already the case that if SoundWire is not enabled in ACPI we just return in all the helpers. The extension is that we return in the helpers when the hardware does not support SoundWire access without the DSP - regardless of what ACPI says.

Reply:
here's the generic code I was referring to:
https://github.com/thesofproject/linux/blob/9b5c7765a81bbd778e2c7b7a32dbe681c00b51d8/sound/soc/sof/intel/hda.c#L968

Reply:
ok no more questions from me then. If you could please address the cosmetic changes mentioned in other comments, it is good for me